Sprint-1/implement/sum.test.js

```diff
@@ -13,24 +13,41 @@ const sum = require("./sum.js");
 // Given an empty array
 // When passed to the sum function
 // Then it should return 0
-test.todo("given an empty array, returns 0")
+test("for an empty array should return 0", () => {
+  expect(sum([])).toBe(0);
+});
 
 // Given an array with just one number
 // When passed to the sum function
 // Then it should return that number
+test("for an array with one number should return that number", () => {
+  expect(sum([4])).toBe(4);
+});
 
 // Given an array containing negative numbers
 // When passed to the sum function
 // Then it should still return the correct total sum
+test("for an array with negative numbers should return sum of them", () => {
+  expect(sum([-5, -9, -4, -10])).toBe(-28);
+});
 
 // Given an array with decimal/float numbers
 // When passed to the sum function
 // Then it should return the correct total sum
+test("for an array with decimal numbers should return sum of them", () => {
+  expect(sum([5.2, 1.9, 8.3])).toEqual(15.4);
+});
 
 // Given an array containing non-number values
 // When passed to the sum function
 // Then it should ignore the non-numerical values and return the sum of the numerical elements
+test("if the array has non-numerical values inside, it should ignore them and just add up the numbers", () => {
+  expect(sum(["hi", 5, 4, 3, "NO"])).toBe(12);
+});
 
 // Given an array with only non-number values
 // When passed to the sum function
 // Then it should return the least surprising value given how it behaves for all other inputs
+test("array with only non-number values will add up the ASCII of the values!", () => {
+  expect(sum(["hi", "bye", "hello"])).toBe(1061);
```

Comment:
Can also include non-number values such as `undefined`, `null`, and `true` in the test.

Reply:
i inserted some more tests to check if the input is undefined, null, and true to return "0"
Thanks